### PR TITLE
Refactoring of notification system

### PR DESCRIPTION
**Pull request by johandahlberg**
I've refactored alot in the notification system to make it more testable, and also to make it easier to extend to other forms of notifiers.

I thought I'd divided this work into two parts, and add the REST notifiers in the next PR to avoid making this a bigger PR than it already is. (In my defense this PR only add 102 lines of code, and most of that should be tests)

### CODE REVIEW

File: src/main/scala/hercules/actors/notifiers/NotifierActor.scala [hunk 1]
```diff
@@ -1,14 +1,65 @@
...
+   * SendNotificationUnitMessage, FailedNotificationUnitMessage
+   */
+  override def receive: Receive = LoggingReceive {
+
+    case failedNotificationUnit: FailedNotificationUnitMessage => {
+      import context.dispatcher
+
+      if (failedNotificationUnit.unit.attempts <= notifierConfig.numRetries)
+        context.system.scheduler.scheduleOnce(
+          notifierConfig.retryInterval.seconds,
```
Comment (b97pla) [inline src/main/scala/hercules/actors/notifiers/NotifierActor.scala]: Should a shutdown hook that cancels the scheduler be added for good practice?

File: src/main/scala/hercules/actors/notifiers/NotifierActor.scala [hunk 2]
```diff
@@ -1,14 +1,65 @@
...
+          self,
+          SendNotificationUnitMessage(failedNotificationUnit.unit))
+      else
+        log.warning(
+          s"Reached maximum number of send retries on ${failedNotificationUnit.unit} " +
+            "will no drop it."
```
Comment (b97pla) [inline src/main/scala/hercules/actors/notifiers/NotifierActor.scala]: typo...

**Comment by b97pla** [comment]
Nice work!

File: src/main/scala/hercules/actors/notifiers/NotifierActor.scala (hunk 1 above)
Comment (johandahlberg) [inline src/main/scala/hercules/actors/notifiers/NotifierActor.scala]: Good point! I will add that.

Comment (johandahlberg) [inline src/main/scala/hercules/actors/notifiers/NotifierActor.scala]: Just to note - I don't know if this is necessary, and if so how to do it. So I'll leave it for now.

**Comment by johandahlberg** [comment]
Ready to merge!

**Pull request merged**